packages/xo-server/src/api/sr.js

```diff
@@ -179,6 +179,43 @@ createIso.resolve = {
   host: ['host', 'host', 'administrate'],
 }
 
+// -------------------------------------------------------------------
+
+export async function createFile({
+  host,
+  nameLabel,
+  nameDescription,
+  location,
+}) {
+  const xapi = this.getXapi(host)
+  const deviceConfig = { location }
+  const srRef = await xapi.call(
+    'SR.create',
+    host._xapiRef,
+    deviceConfig,
+    '0',
+    nameLabel,
+    nameDescription,
+    'file',
+    'user',
+    false,
+    {}
+  )
```

Comment:
Create an `createSr` method in `xapi/mixins/storage` with named params and coherent default values.

Reply:
I have done it, with substitution in `createFile()`, but I don't have the means to validate the other types of SR. For now I have left them out.